File: FloatItPhone/src/main/java/net/geekstools/floatshort/PRO/Utils/LaunchPad/OpenApplicationsLaunchPad.java
/*
 * Copyright © 2020 By Geeks Empire.
 *
 * Created by Elias Fazel on 3/26/20 2:51 PM
 * Last modified 3/26/20 2:32 PM
 *
 * Licensed Under MIT License.
 * https://opensource.org/licenses/MIT
 */

package net.geekstools.floatshort.PRO.Utils.LaunchPad;

import android.app.Activity;
import android.os.Bundle;

import net.geekstools.floatshort.PRO.Utils.Functions.FunctionsClass;

public class OpenApplicationsLaunchPad extends Activity {

    @Override
    protected void onCreate(Bundle Saved) {
        super.onCreate(Saved);

        FunctionsClass functionsClass = new FunctionsClass(getApplicationContext());

        String packageName = getIntent().getStringExtra("packageName");

        try {
            if (getIntent().hasExtra("className")) {

                String className = getIntent().getStringExtra("className");

                functionsClass.
                        openApplicationFromActivity(OpenApplicationsLaunchPad.this,
                                packageName,
                                className);

            } else {

                functionsClass
                        .openApplicationFromActivity(OpenApplicationsLaunchPad.this,
                                packageName);
            }
        } catch (Exception e) {
            e.printStackTrace();

            functionsClass
                    .openApplicationFromActivity(OpenApplicationsLaunchPad.this,
                            packageName);
        }

        finish();
    }
}

File: FloatItPhone/src/main/java/net/geekstools/floatshort/PRO/Utils/LaunchPad/OpenApplicationsLaunchPad.kt
/*
 * Copyright © 2020 By Geeks Empire.
 *
 * Created by Elias Fazel
 * Last modified 5/25/20 7:04 PM
 *
 * Licensed Under MIT License.
 * https://opensource.org/licenses/MIT
 */
package net.geekstools.floatshort.PRO.Utils.LaunchPad

import android.app.Activity
import android.os.Bundle
import net.geekstools.floatshort.PRO.Utils.Functions.FunctionsClass

class OpenApplicationsLaunchPad : Activity() {
    override fun onCreate(Saved: Bundle?) {
        super.onCreate(Saved)

        val functionsClass = FunctionsClass(applicationContext)

        val appPackageName: String? = intent.getStringExtra("packageName")

        appPackageName?.let {

            try {
                if (intent.hasExtra("className")) {

                    val appClassName = intent.getStringExtra("className")

                    functionsClass.openApplicationFromActivity(this@OpenApplicationsLaunchPad,
                            appPackageName,
                            appClassName)

                } else {

                    functionsClass
                            .openApplicationFromActivity(this@OpenApplicationsLaunchPad,
                                    appPackageName)

                }
            } catch (e: Exception) {
                e.printStackTrace()

                functionsClass
                        .openApplicationFromActivity(this@OpenApplicationsLaunchPad,
                                appPackageName)
            }
        }

        this@OpenApplicationsLaunchPad.finish()
    }
}